superset/models/core.py

```diff
@@ -688,22 +698,22 @@ def export_dashboards(cls, dashboard_ids):
             copied_dashboard.alter_params(remote_id=dashboard_id)
             copied_dashboards.append(copied_dashboard)
 
-            eager_datasources = []
-            for datasource_id, datasource_type in datasource_ids:
-                eager_datasource = ConnectorRegistry.get_eager_datasource(
-                    db.session, datasource_type, datasource_id
-                )
-                copied_datasource = eager_datasource.copy()
-                copied_datasource.alter_params(
-                    remote_id=eager_datasource.id,
-                    database_name=eager_datasource.database.name,
-                )
-                datasource_class = copied_datasource.__class__
-                for field_name in datasource_class.export_children:
-                    field_val = getattr(eager_datasource, field_name).copy()
-                    # set children without creating ORM relations
-                    copied_datasource.__dict__[field_name] = field_val
-                eager_datasources.append(copied_datasource)
+        eager_datasources = []
+        for datasource_id, datasource_type in datasource_ids:
+            eager_datasource = ConnectorRegistry.get_eager_datasource(
+                db.session, datasource_type, datasource_id
+            )
+            copied_datasource = eager_datasource.copy()
+            copied_datasource.alter_params(
+                remote_id=eager_datasource.id,
+                database_name=eager_datasource.database.name,
+            )
+            datasource_class = copied_datasource.__class__
+            for field_name in datasource_class.export_children:
+                field_val = getattr(eager_datasource, field_name).copy()
+                # set children without creating ORM relations
+                copied_datasource.__dict__[field_name] = field_val
+            eager_datasources.append(copied_datasource)
```

Comment:
This brings the block out of the outer loop, is it intended/bug?

Reply:
Looks like this was erroneously in the dashboard loop, nothing there connecting them together.

Reply:
I stared at this for a while and couldn't figure out why this block should be in the inner loop - I think it was doing repeat work? Would be great if someone with more context could confirm whether this block should be in the inner loop.